Sealer/sealer.cc

```diff
@@ -603,38 +602,57 @@ void kRun(const ConnectPairsParams& params,
 
 	printLog(logStream, "Flanks inserted into k run = " + IntToString(flanks.size()) + "\n");
 
-	for (read1_it = flanks.begin(); read1_it != flanks.end();) {
-		success = false;
+	int counter = 0;
+	vector<map<FastaRecord, map<FastaRecord, Gap> >::iterator> flanks_closed;
+#pragma omp parallel private(read1_it, read2_it) firstprivate(counter)
```

Comment:
I sit possible to use `#pragma omp parallel for` rather than `counter % omp_get_num_threads() == omp_get_thread_num()`? Is it possible to parallelize a for loop that's iterating over a STL data structure like `std::map`?

Reply:
I'll see if this works, actually. I thought it was not supported, but reading up on it, it seems it should be.

Reply:
It seems comparing map iterators with less than operator (<) is not valid (for the loop exit condition), but it is required for a loop to be parallelized with OpenMP, as it needs a "countable" loop. Map iterators seem to only allow inequality (!=) comparison, which is the case right now.

Reply:
How about storing the keys of the map in a vector, and then `parallel omp for` iterate over that vector?

Reply:
It seems it would be possible to fully replace the map there with a vector in the first place, while building flanks (Actually, is that what you meant?). Otherwise, it might be a bit slow to transfer keys to a vector.

Reply:
Well, that's the better ultimate solution, but it'd make for a bigger diff that's more difficult to review. I meant copy the keys to a second `std:vector` and iterate over that. Let's stick with what you have, using the `counter % omp_get_num_threads() == omp_get_thread_num()`, for this PR. After we merge this PR, I'd appreciate as second PR to change the `std::map` to `std::vector`. In my experience, `std::map` is very rarely the correct data structure to use. Often `std::unordered_map` or a sorted `std::vector` is better.